src/main/java/application/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,52 @@
+package application.repository;
+
+import application.model.Book;
+import java.util.List;
+import java.util.Optional;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
+            try {
+                transaction = session.beginTransaction();
+                session.save(book);

```

Comment:
`save()` is deprecated since [Hibernate 6.0](https://docs.jboss.org/hibernate/orm/6.0/javadocs/org/hibernate/Session.html#save(java.lang.Object)). Use `presist()` instead. 
Make sure to do it everywhere.